core/src/ast/function.rs

```diff
@@ -513,25 +513,67 @@ impl ToSql for Function {
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub enum Aggregate {
-    Count(CountArgExpr),
-    Sum(Expr),
-    Max(Expr),
-    Min(Expr),
-    Avg(Expr),
-    Variance(Expr),
-    Stdev(Expr),
+    Count { expr: CountArgExpr, distinct: bool },
+    Sum { expr: Expr, distinct: bool },
+    Max { expr: Expr, distinct: bool },
+    Min { expr: Expr, distinct: bool },
+    Avg { expr: Expr, distinct: bool },
+    Variance { expr: Expr, distinct: bool },
+    Stdev { expr: Expr, distinct: bool },
```

Comment:
Every enum variant uses `distinct`, then could we consider change `Aggregate` to struct that has aggregate type field and `distinct`?